api/src/test/java/org/apache/iceberg/TestPartitionSpecValidation.java

```diff
@@ -312,4 +314,15 @@ public void testAddPartitionFieldsWithAndWithoutFieldIds() {
     assertThat(spec.fields().get(2).fieldId()).isEqualTo(1006);
     assertThat(spec.lastAssignedFieldId()).isEqualTo(1006);
   }
+
+  @Test
+  public void testVariantUnsupported() {
```

Comment:
What about other transforms? None of them are allowed, right?

Reply:
That's right. I have the coverage in TestIdentity.java and TestBucketing.java and I think I have checked all the transforms.